docs/components/integration-framework/connectors/custom-built-connectors/connector-sdk.md

```diff
@@ -4,17 +4,677 @@ title: Connector SDK
+
+```json
+{
+  "type": "Hidden",
+  "value": "io.camunda:my-connector:1",
+  "binding": {
+    "type": "zeebe:taskDefinition:type"
+  }
+}
+```
+
+This type definition is the connection configuring which version of your connector runtime behavior to use.
+In technical terms, this defines the **Type** of jobs created for tasks in your process model that use this template.
+Consult the [job worker](../../../concepts/job-workers.md) guide to learn more.
+
+Besides the type binding, connector templates also define the input variables of your connector as `zeebe:input` objects.
+For example, you can create the input variable `message` of your connector in the element template as follows:
+
+```json
+{
+  "label": "Message",
+  "type": "Text",
+  "feel": "optional",
+  "binding": {
+    "type": "zeebe:input",
+    "name": "message"
+  }
+}
+```
+
+You can also define nested data structures to reflect domain objects that group attributes.
+For example, you can create the domain object `authentication` that contains the properties
+`user` and `password` as follows:
+
+```json
+{
+  "label": "Username",
+  "description": "The username for authentication.",
+  "type": "String",
+  "binding": {
+    "type": "zeebe:input",
+    "name": "authentication.user"
+  }
+},
+{
+  "label": "Token",
+  "description": "The token for authentication.",
+  "type": "String",
+  "binding": {
+    "type": "zeebe:input",
+    "name": "authentication.token"
+  }
+}
+```
+
+You can easily deserialize these authentication properties into a domain object using the SDK.
+Have a look at the [input data](#input-data) section for further insights.
+
+Connectors that offer any kind of result from their invocation should allow users to configure
+how to map the result into their processes. Therefore, connector templates can reuse the two
+recommended objects, **Result Variable** and **Result Expression**:
+
+```json
+{
+  "label": "Result Variable",
+  "description": "Name of variable to store the response in",
+  "type": "String",
+  "binding": {
+    "type": "zeebe:taskHeader",
+    "key": "resultVariable"
+  }
+},
+{
+  "label": "Result Expression",
+  "description": "Expression to map the response into process variables",
+  "type": "Text",
+  "feel": "required",
+  "binding": {
+    "type": "zeebe:taskHeader",
+    "key": "resultExpression"
+  }
+}
+```
+
+These objects will create custom headers for the jobs created for the tasks that use this template.
+The connector runtime environments will pick up those two custom headers and translate them into process variables accordingly.
+You can see an example of how to use this in the [out-of-the-box REST connector](../out-of-the-box-connectors/rest.md#response).
+
+### Runtime logic
+
+Providing reusable connector runtime behavior is mainly done by implementing and exposing an
+implementation of the `ConnectorFunction` interface of the SDK. The connector runtime
```

Comment:
Minor tweak:
Update text from "Providing reusable connector runtime behavior is mainly done by implementing and exposing..."
to "To create a reusable runtime behavior for your connector, you are required to implement and expose..."

Reply:
Updated